rust/noosphere/src/storage.rs

```diff
@@ -82,3 +62,36 @@ fn get_scoped_path(path: &Path, scope: &Did) -> PathBuf {
 
     path_buf
 }
+
+/// Construct [PlatformStorage] from a [StorageLayout] and [StorageConfig].
+///
+/// Takes a [Url] to an IPFS Gateway that is *required* when
+/// compiling with `ipfs-storage`. Has no effect otherwise.
+pub async fn create_platform_storage(
+    layout: StorageLayout,
+    #[allow(unused)] ipfs_gateway_url: Option<Url>,
+    storage_config: Option<StorageConfig>,
+) -> Result<PlatformStorage> {
+    let path: PathBuf = layout.into();
+    #[cfg(any(sled, rocksdb))]
+    let storage = {
+        use noosphere_storage::ConfigurableStorage;
+        PrimitiveStorage::open_with_config(&path, storage_config.unwrap_or_default()).await?
+    };
+
+    #[cfg(wasm)]
+    let storage = IndexedDbStorage::new(&path.to_string()).await?;
+
+    #[cfg(ipfs_storage)]
+    let storage = {
```

Comment:
This now enables a native build with `ipfs-storage` enabled, and bails if an `ipfs_gateway_url` has not been set as `PlatformStorage`